### PR TITLE
Enhancements to the FHIR Path Tool

### PR DESCRIPTION
**Pull request by carlosfelix2**
This PR makes improvements to the FHIR Path tool.

Test Steps:
1. Check out this branch and build the prime CLI by running `./gradlew fatjar`
2. Run the FHIR path tool by running `./prime fhirpath -i src/testIntegration/resources/datatests/HL7_to_FHIR/sample_co_1_20220518-0001.fhir`
3. Type in the following paths in sequence and verify you get to see the contents of the extension
```
resource: 'Bundle.entry.resource.ofType(MessageHeader)'
%resource.source.extension('https://reportstream.cdc.gov/fhir/StructureDefinition/source-software-vendor-org').value
resource = %resource.source.extension('https://reportstream.cdc.gov/fhir/StructureDefinition/source-software-vendor-org').value
%resource
```

## Changes
- Allow to set the %resource by typing ANY of the following:
```
resource : 'Bundle.entry.resource.ofType(MessageHeader)'
resource = 'Bundle.entry.resource.ofType(MessageHeader)'
resource: Bundle.entry.resource.ofType(MessageHeader)
resource= Bundle.entry.resource.ofType(MessageHeader)
```
- Prints out information about non-root resources, but will not walk deeper into a properties to print out other properties (prints out one level of properties)

### Testing
- [ ] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
Not part of an issue

### CODE REVIEW

File: prime-router/src/main/kotlin/cli/ProcessFhirCommands.kt [hunk 1]
```diff
@@ -151,65 +157,159 @@ class FhirPathCommand : CliktCommand(
...
+    private fun fhirBaseAsString(value: Base): String {
+        return when {
+            value.isPrimitive -> "Primitive: $value"
+
+            // Bundle entries
+            value is BundleEntryComponent -> "Entry: ${value.fullUrl}"
+
+            // References
+            value is Reference ->
+                "Reference to ${value.reference} - set as %resource to navigate into it"
```
Comment (PatriciaPerozoUSDS) [inline prime-router/src/main/kotlin/cli/ProcessFhirCommands.kt]: This is a super helpful message!

File: prime-router/src/main/kotlin/cli/ProcessFhirCommands.kt [hunk 2]
```diff
@@ -151,65 +157,159 @@ class FhirPathCommand : CliktCommand(
...
         echo("", true)
         echo("Using the FHIR bundle in ${inputFile.absolutePath}...", true)
+        echo("Special commands:", true)
+        echo("  resource [= | :] <FHIR Path>   - Sets %resource to a given FHIR path", true)
         echo("Press CTRL-C or ENTER to exit.", true)
 
         // Loop until you press CTRL-C or ENTER at the prompt.
         while (true) {
+            echo("", true)
+            echo("%resource = $focusPath")
```
Comment (PatriciaPerozoUSDS) [inline prime-router/src/main/kotlin/cli/ProcessFhirCommands.kt:177]: This was really helpful in not getting confused about what I had already set as the resource

**Review by PatriciaPerozoUSDS — APPROVED** [review]
Tested this out locally. These changes are super helpful! I don't think this should block this PR but it might be useful to add in an "unset resource" command so that it can quickly go back to the bundle root as the resource without entering `%resource = Bundle` each time.

**Review by TurkeyFried — APPROVED** [review]
Works very well, was able to get to the organization object with the sample code

**Review by PatriciaPerozoUSDS — APPROVED** [review]
The new changes look even better!

**Pull request merged**